feat: ui.progress_bar and ui.progress_circle (#892)
deephaven/deephaven-plugins: plugins/ui/docs/components/progress_circle.md

# Progress Circle

> Progress circles show the progression of a system operation such as downloading, uploading, or processing, in a visual way. They can represent determinate or indeterminate progress.

Author: I don't think so, React Spectrum doesn't say anything either

# Progress Circle

Progress circles show the progression of a system operation such as downloading, uploading, or processing, in a visual way. They can represent determinate or indeterminate progress.

## Example

```python
from deephaven import ui

@ui.component
def ui_progress_circle():
    return ui.progress_circle(size="L", is_indeterminate=True)

progress_circle = ui_progress_circle()
```

## UI Recommendations

> 1. Use the appropriate size based on parent's size

Reviewer: ```suggestion
1. Use the appropriate size based on the parent's size.
```

> 2. Use `static_color="white"` or `static_color="white"` if necessary to make sure the progress circle has enough contrast with the background

Reviewer: the two options are the same

> 2. Use `static_color="white"` or `static_color="white"` if necessary to make sure the progress circle has enough contrast with the background

Reviewer: ```suggestion
2. Use `static_color="white"` or `static_color="white"` if necessary to make sure the progress circle has enough contrast with the background.
```

> 3. If the value of the progress is unknown, use `is_indeterminate=True`

Reviewer: ```suggestion
3. If the value of the progress is unknown, use `is_indeterminate=True`.
```

## Visual Options

> Progress Circle comes in 3 different sizes by the `size` prop: `"S"`, `"M"`, and `"L"`. Furthermore, the `static_color` prop can be used to control the color of the progress circle.

Reviewer: ```suggestion
Progress Circle comes in three different sizes determined by the `size` prop: `"S"`, `"M"`, and `"L"`. Furthermore, the `static_color` prop can be used to control the color of the progress circle.
```

# Progress Circle

Progress circles show the progression of a system operation such as downloading, uploading, or processing, in a visual way. They can represent determinate or indeterminate progress.

## Example

```python
from deephaven import ui

@ui.component
def ui_progress_circle():
    return ui.progress_circle(size="L", is_indeterminate=True)

progress_circle = ui_progress_circle()
```

## UI Recommendations

1. Use the appropriate size based on the parent's size.
2. Use `static_color="white"` or `static_color="black"` if necessary to make sure the progress circle has enough contrast with the background.
3. If the value of the progress is unknown, use `is_indeterminate=True`.

## Value 

The progress is controlled by the `value`, `min_value`, and `max_value` props. The default values of `min_value` and `max_value` are `0` and `100`, respectively. 

```python
def value_variants():
    return ui.flex(
        ui.progress_circle(value=50),
        ui.progress_circle(value=50, min_value=25, max_value=125),

>         column_gap="20px",

Reviewer 2: there's already default gap, this can be dropped.

```suggestion
```

    )

progress_circle_value_examples = value_variants()
```

## Indeterminate

Use `is_indeterminate=True` if the progress can not be determined.

```python
def indeterminate_variants():
    return ui.flex(
        ui.progress_circle(value=70),
        ui.progress_circle(is_indeterminate=True),
        column_gap="20px",
    )

progress_circle_indeterminate_examples = indeterminate_variants()
```

## Size

Progress Circle comes in three different sizes determined by the `size` prop: `"S"`, `"M"`, and `"L"`. By default, the size is `"M"`.

```python
def size_variants():
    return ui.flex(
        ui.progress_circle(value=70, size="S"),
        ui.progress_circle(value=70),
        ui.progress_circle(value=70, size="L"),

>         column_gap="20px",

Reviewer 2: ```suggestion
```

    )

progress_circle_size_examples = size_variants()
```

## Static Color

The `static_color` prop can be used to control the color of the progress circle between the default color, `"black"`, and `"white"`.

```python
def color_variants():
    return ui.view(
        ui.flex(
            ui.view(ui.progress_circle(value=70, margin="10px")),

>             ui.view(ui.progress_circle(value=70, static_color="white", margin="10px")),

Reviewer 2: add background_color="black" to this example so it's static. 

Light theme users exist.

![image](https://github.com/user-attachments/assets/99949296-7d53-472e-92fd-db9718cbc94d)
